src/components/SigningModal/TransactionStepper/TransactionStepper.js

```diff
@@ -49,35 +57,52 @@ function TransactionStepper({ steps, onComplete, className }) {
 
   const stepsCount = steps.length - 1
 
-  const renderStep = (stepIndex, showDivider) => {
-    const title = steps[stepIndex][0]
-    const { status, hash } = stepState[stepIndex]
+  const renderStep = useCallback(
+    (stepIndex, showDivider) => {
+      const { title, descriptions } = steps[stepIndex]
+      const { status, hash } = stepState[stepIndex]
 
-    return (
-      <li
-        key={stepIndex}
-        css={`
-          display: flex;
-        `}
-      >
-        <Step
-          title={title}
-          number={stepIndex + 1}
-          status={status}
-          showDivider={showDivider}
-          transactionHash={hash}
-        />
-      </li>
-    )
-  }
+      const { waiting, prompting, working, success, error } =
+        descriptions || DEFAULT_DESC
+
+      const descWithDefault = {
+        [STEP_WAITING]: waiting || DEFAULT_DESC['waiting'],
+        [STEP_PROMPTING]: prompting || DEFAULT_DESC['prompting'],
+        [STEP_WORKING]: working || DEFAULT_DESC['working'],
+        [STEP_SUCCESS]: success || DEFAULT_DESC['success'],
+        [STEP_ERROR]: error || DEFAULT_DESC['error'],
+      }
+
+      const desc = descWithDefault[status]
```

Comment:
This is the only part that I find a bit weird; could we use the symbol directly in the description object?

That way we might be able to do an object merge or `const desc = descriptions[status] || DEFAULT_DESC[status]`.

Reply:
Totes. I nearly went with this originally but landed on not requiring the imports. This makes things easier though 👍